README.md

```diff
@@ -49,8 +51,181 @@ If you would prefer to run the rebuild steps manually instead of using the above
+* **M3U Play List URL**: If location is `Remote path` this setting must contain a valid URL for the addon to function.
+* **Cache M3U at local storage**: If location is `Remote path` select whether or not the the M3U file should be cached locally.
+* **Numbering channel starts at**: The number to start numbering channels from.
+
+### EPG Settings
+Settings related to the EPG.
+
+My default the addon will read the first `<category>` element of a `programme` and use this as the genre string. It is also possible to supply a mapping file to convert the genre string to a genre ID, allowing colour coding of the EPG. Please see: [Using a mapping file for Genres](#using-a-mapping-file-for-genres) in the Appendix for details on how to set this up.
+
+* **Location**: Select where to find the XMLTV resource. The options are:
+    - `Local path` - A path to an XMLTV file whether it be on the device or the local network.
+    - `Remote path` - A URL specifying the location of the XMLTV file.
+* **XMLTV Path**: If location is `Local Path` this setting should contain a valid path.
+* **XMLTV URL**: If location is `Remote Path` this setting should contain a valid URL.
+* **Cache XMLTV at local storage**: If location is `Remote path` select whether or not the the XMLTV file should be cached locally.
+* **EPG Time Shift (hours)**: Adjust the EPG times by this value in minutes, range is from -12 hours to +14 hours.
+* **Apply Time Shift To All Channels**: Whether or not to override the time shift for all channels with `EPG time shift`. If not enabled `EPG time shift` plus the individual time shift per channel (if available) will be used.
+
+### Channel Logos
+Settings realted to Channel Logos.
+
+* **Location**: Select where to find the channel logos. The options are:
+    - `Local path` - A path to a folder whether it be on the device or the local network.
+    - `Remote path ` - A base URL specifying the location of the logos.
+* **Channel Logos Folder**: If location is `Local Path` this setting should contain a valid folder.
+* **Channel Logos Base URL**: If location is `Remote Path` this setting should contain a valid base URL.
+* **Channel Logos from XMLTV**: Preference on how to handle channel logos. The options are:
+    - `Ignore` - Don't use channel logos from an XMLTV file.
+    - `Prefer M3U` - Use the channel logo from the M3U if available otherwise use the XMLTV logo.
+    - `Prefer XMLTV` - Use the channel logo from the XMLTV file if available otherwise use the M3U logo.
+
 ## Appendix
 
+### Using a mapping file for Genres
+
+Users can create there own genre mapping files to map their genre strings to genre IDs. This allows the EPG UI to be colour coded per genre.
+
+Kodi uses the following standard for it's genre IDs: https://www.etsi.org/deliver/etsi_en/300400_300499/300468/01.11.01_60/en_300468v011101p.pdf
+
+The file created must be called `genres.xml` and be located here: `userdata/addon_data/pvr.iptvsimple`.
```

Comment:
@phunkyfish could this perhaps be an option to specify within the settings similar to M3U and XMLTV settings? That way it can be served through a server if necessary.

Reply:
Yes, but will be a feature in Matrix, not the Leia branch.

Reply:
The above was already there. I just documented it.

Reply:
Aah okay! Thanks!

Reply:
if need here is full worked [genres.xml](http://bit.ly/2z5z362) with types, subtype in English

Reply:
There is actually one there in Matrix PR except it uses hex instead of type and subtype for the  genre ID. In Matrix it will support a single hex value or two integers. Up to the user.

Reply:
Will add yours as an alternative using type and subtype

Reply:
hm, maybe i don't understand, but would be nice feature, if can be describe in xmltv file
```xml
<category type="16" subtype="16">Action / Comedy</category>
```
and kodi will show color for movie (blue) and write not default name 'Movie / Drama' but  use from xmltv name 'Action / Comedy'
would be nice feature for sport, because can be describe genre type and name
```xml
<category type="64" subtype="16">Formula 1</category>
<category type="64" subtype="16">Basketball</category>
```
`subtype="16"` is user defined 

for this need to be change:
https://github.com/kodi-pvr/pvr.iptvsimple/blob/2d9bf65a116d97670bd0bf9c24cee591817f3e5b/src/PVRIptvData.cpp#L1054-L1064
and add to read genre type subtype
https://github.com/kodi-pvr/pvr.iptvsimple/blob/2d9bf65a116d97670bd0bf9c24cee591817f3e5b/src/PVRIptvData.cpp#L420

Reply:
Won't work, that would need to change in kodi. If you set a genre type and subtype you lose the custom text.

Reply:
But it can work in Matrix: https://github.com/xbmc/xbmc/pull/16541
!!!